src/net/discovery/config.rs

```diff
@@ -0,0 +1,145 @@
+/// query parameters
+#[derive(Debug, PartialEq, Eq)]
+struct Origin(Scheme, Authority);
+
+impl Origin {
+    fn to_uri(&self) -> Uri {
+        Uri::builder()
+            .scheme(self.0.clone())
+            .authority(self.1.clone())
+            .path_and_query("")
+            .build()
+            .unwrap()
+    }
+}
+
+impl<'de> Deserialize<'de> for Origin {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        struct UrlVisitor;
+        impl<'de> serde::de::Visitor<'de> for UrlVisitor {
+            type Value = Origin;
+
+            fn expecting(&self, formatter: &mut Formatter) -> std::fmt::Result {
+                formatter.write_str("a valid format origin made up of http[s]://<authority>")
+            }
+
+            fn visit_str<E: serde::de::Error>(self, v: &str) -> Result<Self::Value, E> {
+                let uri = v.parse::<Uri>().map_err(|err| E::custom(err.to_string()))?;
+                let parts = uri.into_parts();
+                let scheme = parts.scheme.ok_or_else(|| E::custom("missing scheme"))?;
+                let authority = parts
+                    .authority
+                    .ok_or_else(|| E::custom("missing authority"))?;
+                Ok(Origin(scheme, authority))
+            }
+        }
+        deserializer.deserialize_str(UrlVisitor)
+    }
+}
+
+/// Configuration values for a single peer helper of the MPC ring
+#[derive(Deserialize)]
+struct HelperConfig {
+    origin: Origin,
+}
+
+/// All config value necessary to discover all peer helpers of the MPC ring
+#[derive(Deserialize)]
+struct Config {
+    h1: HelperConfig,
+    h2: HelperConfig,
+    h3: HelperConfig,
+}
+
+impl Config {
+    /// Reads config from `file_location`. Expects file to be json format
+    /// # Errors
+    /// if the file does not exist, or is in an invalid format
+    #[allow(dead_code)] // TODO: will use in upcoming PR
+    pub fn new(file_location: PathBuf) -> Result<Self, Error> {
+        use std::{fs::File, io::BufReader};
+
+        let contents = File::open(file_location)?;
+        let buffered = BufReader::new(contents);
+        Ok(serde_json::from_reader(buffered)?)
+    }
+}
+
+impl PeerDiscovery for Config {
+    fn peers(&self) -> [MpcHttpConnection; 3] {
+        let h1 = MpcHttpConnection::new(self.h1.origin.to_uri());
+        let h2 = MpcHttpConnection::new(self.h2.origin.to_uri());
+        let h3 = MpcHttpConnection::new(self.h3.origin.to_uri());
+        [h1, h2, h3]
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use rand::distributions::Alphanumeric;
+    use rand::{thread_rng, Rng};
+    use std::fs::File;
+    use std::io::Write;
+
+    const H1_URI: &str = "http://localhost:3000";
+    const H2_URI: &str = "http://localhost:3001";
+    const H3_URI: &str = "http://localhost:3002";
+    const EXAMPLE_CONFIG: &str = r#"
```

Comment:
I am not a huge fan of json configs - they are generally hard to read and follow. also I would expect us to need some complex configuration management, so shall we talk to Martin about importing https://docs.rs/config/latest/config/ and using `toml`?

Reply:
I'm definitely on board for that